### ISSUE TITLE
(Epic) Define semantics (and testing strategy) when SDK version is revved

### ISSUE THREAD

**Issue by MonsieurNicolas** [stellar/rs-soroban-sdk#1095]
This is similar to stellar/rs-soroban-env#310 but scenarios are around SDK changes that are not impacting the core protocol.

Context:
* contract `A` and `B` are compiled with SDK 1.0.0
* contract `A` depends on `B` (ie: `A` invokes `B`)

Scenario:
* `A`'s developer wants to upgrade to SDK 2.0.0 and wants to test what happens when calling `B` compiled with version 1.0.0
* `B` gets recompiled using SDK 2.0.0, `A`'s developer wants to test and understand how `A` would be impacted.

This is somewhat related to stellar/rs-soroban-env#289 

```[tasklist]
### Tasks
- [ ] https://github.com/stellar/rs-soroban-sdk/issues/1082
- [ ] https://github.com/stellar/soroban-docs/issues/561
```

**Comment by leighmcculloch** [comment]
Adding functionality to the SDK to make it trivial to test contract consistency over time:
- https://github.com/stellar/rs-soroban-sdk/pull/1137

**Comment by leighmcculloch** [comment]
With #1137 implemented I think we're providing a lot for developers to use to detect unexpected inconsistencies in their contracts. The main two features we're providing is:
- Integration testing of .wasm across SDK/env/protocol versions. (Existing functionality already in the SDK.)
- Recording behavior for tracking over time. (Implemented in #1137)

The last thing missing is documentation to capture these important concepts, which will be addressed by:
- https://github.com/stellar/soroban-docs/issues/561

**Comment by anupsdf** [comment]
Closing this because its done and only doc issue is pending.
